File: WebAPI/Models/Dtos/GetListDto.cs
namespace WebAPI.Models.Dtos
{
    public class GetListDto
    {
        public DateTime Date { get; set; }
        public CalendarView CalendarView { get; set; }
        public bool IgnoreCompleted { get; set; }
        public string? ListTypeName { get; set; } = null;
    }
}

File: WebApp/src/app/models/getListDto.ts
import { CalendarView } from './global/enums';

export class GetListDto {
  date: Date;
  calendarView: CalendarView;
  ignoreCompleted: boolean;
  listTypeName: string | null;

  constructor(obj?: GetListDto) {
    this.date = obj?.date || new Date();
    this.calendarView = obj?.calendarView || CalendarView.Month;
    this.ignoreCompleted = obj?.ignoreCompleted || false;
    this.listTypeName = obj?.listTypeName || null;
  }
}
